src/renderers/shared/stack/reconciler/ReactChildReconciler.js

```diff
@@ -22,11 +22,7 @@ var warning = require('fbjs/lib/warning');
 
 var ReactComponentTreeHook;
 
-if (
-  typeof process !== 'undefined' &&
-  process.env &&
-  process.env.NODE_ENV === 'test'
-) {
```

Comment:
Yes, please revert. I don’t see us fixing this in 15 given that the hack was added to work around an accidental breaking change in 15.2.1: https://github.com/facebook/react/issues/7240. Let’s wait for 16 here please.